core/http-auth-spi/src/main/java/software/amazon/awssdk/http/auth/spi/internal/DefaultNoAuthAuthScheme.java

```diff
@@ -0,0 +1,129 @@
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License").
+ * You may not use this file except in compliance with the License.
+ * A copy of the License is located at
+ *
+ *  http://aws.amazon.com/apache2.0
+ *
+ * or in the "license" file accompanying this file. This file is distributed
+ * on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ * express or implied. See the License for the specific language governing
+ * permissions and limitations under the License.
+ */
+
+package software.amazon.awssdk.http.auth.spi.internal;
+
+import java.nio.ByteBuffer;
+import java.util.Optional;
+import java.util.concurrent.CompletableFuture;
+import org.reactivestreams.Publisher;
+import software.amazon.awssdk.annotations.SdkInternalApi;
+import software.amazon.awssdk.http.ContentStreamProvider;
+import software.amazon.awssdk.http.SdkHttpRequest;
+import software.amazon.awssdk.http.auth.spi.AsyncSignRequest;
+import software.amazon.awssdk.http.auth.spi.AsyncSignedRequest;
+import software.amazon.awssdk.http.auth.spi.HttpSigner;
+import software.amazon.awssdk.http.auth.spi.IdentityProviderConfiguration;
+import software.amazon.awssdk.http.auth.spi.NoAuthAuthScheme;
+import software.amazon.awssdk.http.auth.spi.SyncSignRequest;
+import software.amazon.awssdk.http.auth.spi.SyncSignedRequest;
+import software.amazon.awssdk.identity.spi.IdentityProvider;
+import software.amazon.awssdk.identity.spi.ResolveIdentityRequest;
+
+/**
+ * A default, no-op implementation of {@link NoAuthAuthScheme}. This implementation always:
+ *
+ * <ul>
+ *     <li>Returns an {@link IdentityProvider} that always returns the same static instance that implements the
+ *     {@link AnonymousIdentity} interface</li>
+ *     <li>Returns an {@link HttpSigner} that returns the same request given in the signing request.</li>
+ * </ul>
+ */
+@SdkInternalApi
+public class DefaultNoAuthAuthScheme implements NoAuthAuthScheme {
+    private static final DefaultNoAuthAuthScheme DEFAULT = new DefaultNoAuthAuthScheme();
+    private static final IdentityProvider<AnonymousIdentity> DEFAULT_IDENTITY_PROVIDER = noAuthIdentityProvider();
+    private static final HttpSigner<AnonymousIdentity> DEFAULT_SIGNER = noAuthSigner();
+    private static final AnonymousIdentity ANONYMOUS_IDENTITY = anonymousIdentity();
+
+    /**
+     * Returns an instance of the {@link NoAuthAuthScheme}.
+     */
+    public static NoAuthAuthScheme create() {
+        return DEFAULT;
+    }
+
+    @Override
+    public String schemeId() {
+        return SCHEME_ID;
+    }
+
+    @Override
+    public IdentityProvider<AnonymousIdentity> identityProvider(IdentityProviderConfiguration providers) {
+        return DEFAULT_IDENTITY_PROVIDER;
+    }
+
+    @Override
+    public HttpSigner<AnonymousIdentity> signer() {
+        return DEFAULT_SIGNER;
+    }
+
+    private static IdentityProvider<AnonymousIdentity> noAuthIdentityProvider() {
```

Comment:
**NIT**: it is a bit cumbersome to define a static method that returns an anonymous class... is there a particular reason for that? Why not just use straight up a private static class:

```
private static class NoAuthIdentityProvider implements IdentityProvider<AnonymousIdentity> {
// ...
}
```
and have 
```
private static final IdentityProvider<AnonymousIdentity> DEFAULT_IDENTITY_PROVIDER = new NoAuthIdentityProviderI()
```